### PR TITLE
feat: add support for none config in dotfolder

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
feat

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
NA

**Summary**
- Add support for reading none config from dotfolder
- Add relevant tests

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

**Comment by anshumanv** [comment]
> Can we add tests where we have multiple configuration and load them based on mode?

Config resolves doesn't respect mode, will add the support

File: packages/webpack-cli/lib/groups/ConfigGroup.js [hunk 1]
```diff
@@ -4,14 +4,25 @@ const { extensions } = require('interpret');
...
     '.webpack/webpackfile',
+    'webpack.config.development',
     'webpack.config',
+    'webpack.config.production',
 ];
 
+const modeAlias = {
+    production: 'prod',
+    development: 'dev',
+};
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/ConfigGroup.js:24]: What about none?

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:24]: none has no alias, no need

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:24]: It matches either mode or it's alias, none will match the mode, no need alias

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:24]: @evilebottnawi

**Comment by anshumanv** [comment]
Yes, in near future @evilebottnawi 👍

**Comment by anshumanv** [comment]
@evilebottnawi added 👍

**Comment by anshumanv** [comment]
> So we prefer webpack.config.develoment.js over .webpack/webpack.config.development.js?


Yes

**Comment by evilebottnawi** [comment]
@webpack/cli-team we need small discussion about it, should we prefer config from `.webpack` directory over root configuration or not?

**Comment by rishabh3112** [comment]
`.webpack/*` as this now what we intend to be the default directory for configurations by design.

**Comment by anshumanv** [comment]
Seems fair

**Comment by anshumanv** [comment]
Updated

File: packages/webpack-cli/lib/groups/ConfigGroup.js (hunk 1 above)
Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:26]: Can you add comment how we prefer configurations here for future developers?

**Pull request merged**